pkg/kubelet/cm/devicemanager/manager.go

```diff
@@ -1068,6 +1068,21 @@ func (m *ManagerImpl) isDevicePluginResource(resource string) bool {
 	return false
 }
 
+// GetAllocatableDevices returns information about all the devices known to the manager
+func (m *ManagerImpl) GetAllocatableDevices() map[string]map[string]pluginapi.Device {
+	m.mutex.Lock()
+	defer m.mutex.Unlock()
+	resp := make(map[string]map[string]pluginapi.Device)
+	for resourceName, resourceDevs := range m.allDevices {
+		resp[resourceName] = make(map[string]pluginapi.Device)
+		for devID, dev := range resourceDevs {
+			resp[resourceName][devID] = dev
+		}
+	}
+	klog.V(4).Infof("known devices: %d", len(resp))
```

Comment:
Is this just making a copy of `allDevices`?
If we put a type around  `map[string]map[string]pluginapi.Device` we can just create a `Clone()` call on it to simplify this here.

Reply:
good point, changing as suggested